manim/animation/transform.py

```diff
@@ -401,3 +403,135 @@ def interpolate(self, alpha: float) -> None:
         self.start_anim.interpolate(alpha)
         self.end_anim.interpolate(alpha)
         Transform.interpolate(self, alpha)
+
+
+class FadeTransform(Transform):
+    """Fades one mobject into another.
+
+    Parameters
+    ----------
+    mobject
+        The starting :class:`~.Mobject`.
+    target_mobject
+        The target :class:`~.Mobject`.
+    stretch
+        Controls whether the target :class:`~.Mobject` is stretched during
+        the animation. Default: ``True``.
+    dim_to_match
+        If the target mobject is not stretched automatically, this allows
+        to adjust the initial scale of the target :class:`~.Mobject` while
+        it is shifted in. Setting this to 0, 1, and 2, respectively,
+        matches the length of the target with the length of the starting
+        :class:`~.Mobject` in x, y, and z direction, respectively.
+    kwargs
+        Further keyword arguments are passed to the parent class.
+
+    Examples
+    --------
+
+    .. manim:: DifferentFadeTransforms
+
+        class DifferentFadeTransforms(Scene):
+            def construct(self):
+                starts = [Rectangle(width=4, height=1) for _ in range(3)]
+                VGroup(*starts).arrange(DOWN, buff=1).shift(3*LEFT)
+                targets = [Circle(fill_opacity=1).scale(0.25) for _ in range(3)]
+                VGroup(*targets).arrange(DOWN, buff=1).shift(3*RIGHT)
+
+                self.play(*[FadeIn(s) for s in starts])
+                self.play(
+                    FadeTransform(starts[0], targets[0], stretch=True),
+                    FadeTransform(starts[1], targets[1], stretch=False, dim_to_match=0),
+                    FadeTransform(starts[2], targets[2], stretch=False, dim_to_match=1)
+                )
+
+                self.play(*[FadeOut(mobj) for mobj in self.mobjects])
+
+    """
+
+    def __init__(self, mobject, target_mobject, stretch=True, dim_to_match=1, **kwargs):
```

Comment:
Can you type this code or at least have typings in docstrings? IIRC, all the code in `animation` is already typed.

Reply:
Done via 9cdf699cfff99ac166f901cfd0f71a674668b631.